src/nvidia_rag/rag_server/vlm.py

```
@@ -897,28 +925,57 @@ async def stream_with_messages(
+
+                    if filter_enabled:
+                        # Only the final answer reaches the client
+                        if content:
+                            yield content
+                    else:
+                        # Both reasoning trace and final answer reach the client
+                        if reasoning:
+                            yield reasoning
+                        if content:
+                            yield content
                 except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)